### PR TITLE
Add isapprox for Segments

### PR DESCRIPTION
**Pull request by HaoxuanGuo**
Add isapprox for Segments

### CODE REVIEW

File: src/polytopes/segment.jl [hunk 1]
```diff
@@ -41,3 +41,16 @@ function Base.in(p::Point{Dim,T}, s::Segment{Dim,T}) where {Dim,T}
...
+    isapprox(s1::Segment, s2::Segment)
+
+Tells whether or not the coordinates of segments `s1` and `s2 are approximately equal.
+"""
+function Base.isapprox(s1::Segment{Dim,T}, s2::Segment{Dim,T}) where {Dim,T}
+  function reverse(s::Segment{Dim,T})
+    Segment(s.vertices[2], s.vertices[1])
+  end
+  return all(isapprox.(s1.vertices, s2.vertices)) ||
+    all(isapprox.(reverse(s1).vertices, s2.vertices))
```
Comment (juliohm) [inline src/polytopes/segment.jl]: Segment are oriented, we only want the result to be true if the vertices match in the exact same order.

**Review by juliohm — CHANGES_REQUESTED** [review]

**Comment by GuoHaoxuan** [comment]
@juliohm Fixed oriented.

File: src/polytopes/segment.jl [hunk 2]
```diff
@@ -41,3 +41,12 @@ function Base.in(p::Point{Dim,T}, s::Segment{Dim,T}) where {Dim,T}
...
   iscollinear(a, b, p) && zero(T) ≤ ab ⋅ ap ≤ ab ⋅ ab
 end
+
+"""
+    isapprox(s1::Segment, s2::Segment)
+
+Tells whether or not the coordinates of segments `s1` and `s2 are approximately equal.
+"""
+function Base.isapprox(s1::Segment{Dim,T}, s2::Segment{Dim,T}) where {Dim,T}
+  return all(isapprox.(s1.vertices, s2.vertices))
```
Comment (juliohm) [inline src/polytopes/segment.jl]:
```suggestion
function Base.isapprox(s1::Segment, s2::Segment)
  v1 = s1.vertices
  v2 = s2.vertices
  isapprox(v1[1], v2[1]) && isapprox(v1[2], v2[2])
```

File: src/polytopes/segment.jl [hunk 3]
```diff
@@ -41,3 +41,14 @@ function Base.in(p::Point{Dim,T}, s::Segment{Dim,T}) where {Dim,T}
   ab, ap = b - a, p - a
   iscollinear(a, b, p) && zero(T) ≤ ab ⋅ ap ≤ ab ⋅ ab
 end
+
+"""
+    isapprox(s1::Segment, s2::Segment)
+
+Tells whether or not the coordinates of segments `s1` and `s2 are approximately equal.
```
Comment (juliohm) [inline src/polytopes/segment.jl]:
```suggestion
Tells whether or not the segments `s1` and `s2 are approximately equal.
```

File: src/polytopes/segment.jl [hunk 4]
```diff
@@ -41,3 +41,14 @@ function Base.in(p::Point{Dim,T}, s::Segment{Dim,T}) where {Dim,T}
   ab, ap = b - a, p - a
   iscollinear(a, b, p) && zero(T) ≤ ab ⋅ ap ≤ ab ⋅ ab
 end
+
+"""
+    isapprox(s1::Segment, s2::Segment)
+
+Tells whether or not the segments `s1` and `s2 are approximately equal.
+"""
```
Comment (juliohm) [inline src/polytopes/segment.jl]:
```suggestion
```

File: test/polytopes.jl [hunk 5]
```diff
@@ -6,6 +6,9 @@
     s = Segment(Point(1.0,), Point(2.0,))
     @test all(Point(x,) ∈ s for x in 1:0.01:2)
     @test all(Point(x,) ∉ s for x in [-1.0, 0.0, 0.99, 2.1, 5.0, 10.0])
+    @test s ≈ s
+    @test !(s ≈ Segment(Point(2.0,), Point(1.0,)))
+    @test !(s ≈ Segment(Point(-1.0,), Point(2.0,)))
```
Comment (juliohm) [inline test/polytopes.jl]: Please use P2 instead of Point

File: test/polytopes.jl [hunk 6]
```diff
@@ -6,6 +6,9 @@
     s = Segment(Point(1.0,), Point(2.0,))
```
Comment (juliohm) [inline test/polytopes.jl]:
```suggestion
    s = Segment(P2(1.0,), P2(2.0,))
```

File: test/polytopes.jl [hunk 7]
```diff
@@ -3,9 +3,12 @@
     @test paramdim(Segment) == 1
     @test nvertices(Segment) == 2
 
-    s = Segment(Point(1.0,), Point(2.0,))
+    s = Segment(P2(1.0,), P2(2.0,))
     @test all(Point(x,) ∈ s for x in 1:0.01:2)
     @test all(Point(x,) ∉ s for x in [-1.0, 0.0, 0.99, 2.1, 5.0, 10.0])
+    @test s ≈ s
+    @test !(s ≈ Segment(Point(2.0,), Point(1.0,)))
+    @test !(s ≈ Segment(Point(-1.0,), Point(2.0,)))
```
Comment (juliohm) [inline test/polytopes.jl]:
```suggestion
    @test !(s ≈ Segment(P2(2.0,), P2(1.0,)))
    @test !(s ≈ Segment(P2(-1.0,), P2(2.0,)))
```

File: test/polytopes.jl [hunk 8]
```diff
@@ -17,14 +20,23 @@
...
+    @test s ≈ s
+    @test !(s ≈ Segment(P3(1,1,1), P3(0,0,0)))
+    @test !(s ≈ Segment(P3(1,1,1), P3(0,1,0)))
     s = Segment(Point(1.,1.,1.,1.), Point(2.,2.,2.,2.))
     @test all(Point(x, x, x, x) ∈ s for x in 1:0.01:2)
     @test all(p ∉ s for p in [Point(0.99, 0.99, 0.99, 0.99), Point(2.1, 2.1, 2.1, 2.1)])
     @test all(p ∉ s for p in [Point(1.5, 1.5, 1.5, 1.49), Point(1, 1, 2, 1.0)])
+    @test s ≈ s
+    @test !(s ≈ Segment(Point(2,2,2,2), Point(1,1,1,1)))
+    @test !(s ≈ Segment(Point(1,1,2,1), Point(0,0,0,0)))
```
Comment (juliohm) [inline test/polytopes.jl:39]:
```suggestion
    @test !(s ≈ Segment(P2(2,2,2,2), P2(1,1,1,1)))
    @test !(s ≈ Segment(P2(1,1,2,1), P2(0,0,0,0)))
```

File: test/polytopes.jl (hunk 6 above)
Comment (hx-guo) [reply test/polytopes.jl]: @juliohm This is not a `P2`. This is a 1D segment.

File: test/polytopes.jl [hunk 9]
```diff
@@ -3,9 +3,12 @@
     @test paramdim(Segment) == 1
     @test nvertices(Segment) == 2
 
-    s = Segment(Point(1.0,), Point(2.0,))
+    s = Segment(P2(1.0,), P2(2.0,))
```
Comment (juliohm) [inline test/polytopes.jl]:
```suggestion
    s = Segment(P1(1.0,), P1(2.0,))
```

File: test/polytopes.jl [hunk 10]
```diff
@@ -3,9 +3,12 @@
     @test paramdim(Segment) == 1
     @test nvertices(Segment) == 2
 
-    s = Segment(Point(1.0,), Point(2.0,))
+    s = Segment(P1(1.0,), P1(2.0,))
     @test all(Point(x,) ∈ s for x in 1:0.01:2)
     @test all(Point(x,) ∉ s for x in [-1.0, 0.0, 0.99, 2.1, 5.0, 10.0])
+    @test s ≈ s
+    @test !(s ≈ Segment(P2(2.0,), P2(1.0,)))
+    @test !(s ≈ Segment(P2(-1.0,), P2(2.0,)))
```
Comment (juliohm) [inline test/polytopes.jl]:
```suggestion
    @test !(s ≈ Segment(P1(2.0,), P1(1.0,)))
    @test !(s ≈ Segment(P1(-1.0,), P1(2.0,)))
```

File: test/polytopes.jl [hunk 11]
```diff
@@ -17,14 +20,23 @@
...
+    @test s ≈ s
+    @test !(s ≈ Segment(P3(1,1,1), P3(0,0,0)))
+    @test !(s ≈ Segment(P3(1,1,1), P3(0,1,0)))
     s = Segment(Point(1.,1.,1.,1.), Point(2.,2.,2.,2.))
     @test all(Point(x, x, x, x) ∈ s for x in 1:0.01:2)
     @test all(p ∉ s for p in [Point(0.99, 0.99, 0.99, 0.99), Point(2.1, 2.1, 2.1, 2.1)])
     @test all(p ∉ s for p in [Point(1.5, 1.5, 1.5, 1.49), Point(1, 1, 2, 1.0)])
+    @test s ≈ s
+    @test !(s ≈ Segment(P2(2,2,2,2), P2(1,1,1,1)))
+    @test !(s ≈ Segment(P2(1,1,2,1), P2(0,0,0,0)))
```
Comment (juliohm) [inline test/polytopes.jl]:
```suggestion
    @test !(s ≈ Segment(Point(2,2,2,2), Point(1,1,1,1)))
    @test !(s ≈ Segment(Point(1,1,2,1), Point(0,0,0,0)))
```

File: test/polytopes.jl [hunk 12]
```diff
@@ -3,9 +3,12 @@
     @test paramdim(Segment) == 1
     @test nvertices(Segment) == 2
 
-    s = Segment(Point(1.0,), Point(2.0,))
+    s = Segment(P1(1.0,), P1(2.0,))
     @test all(Point(x,) ∈ s for x in 1:0.01:2)
     @test all(Point(x,) ∉ s for x in [-1.0, 0.0, 0.99, 2.1, 5.0, 10.0])
```
Comment (juliohm) [inline test/polytopes.jl]:
```suggestion
    @test all(P1(x,) ∈ s for x in 1:0.01:2)
    @test all(P1(x,) ∉ s for x in [-1.0, 0.0, 0.99, 2.1, 5.0, 10.0])
```

**Pull request merged**